modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/StartCachesInParallelTest.java

```diff
@@ -0,0 +1,123 @@
+ * Tests, that cluster could start and activate with all possible values of IGNITE_ALLOW_START_CACHES_IN_PARALLEL.
+ */
+public class StartCachesInParallelTest extends GridCommonAbstractTest {
+
+    /** IGNITE_ALLOW_START_CACHES_IN_PARALLEL option value before tests. */
+    private String allowParallel;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setDataStorageConfiguration(
+            new DataStorageConfiguration()
+                .setDefaultDataRegionConfiguration(new DataRegionConfiguration().setPersistenceEnabled(true))
+                .setCheckpointFrequency(500L));
+
+        cfg.setCacheConfiguration(
+            new CacheConfiguration<>()
+                .setName(DEFAULT_CACHE_NAME)
+                .setIndexedTypes(Integer.class, Integer.class));
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        allowParallel = System.getProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        super.afterTestsStopped();
+
+        if (allowParallel != null)
+            System.setProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL, allowParallel);
+        else
+            System.clearProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        System.setProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL, "false");
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** */
+    public void testWithEnabledOption() throws Exception {
+        doTest("true");
+    }
+
+    /** */
+    public void testWithDisabledOption() throws Exception {
+        doTest("false");
+    }
+
+    /** */
+    public void testWithoutOption() throws Exception {
+        doTest(null);
+    }
+
+    /**
+     * Test routine.
+     *
+     * @param optionValue IGNITE_ALLOW_START_CACHES_IN_PARALLEL value.
+     * @throws Exception If failed.
+     */
+    private void doTest(String optionValue) throws Exception {
+        if (optionValue == null)
+            System.clearProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL);
+        else {
+            Boolean.parseBoolean(optionValue);
+
+            System.setProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL, optionValue);
+        }
+
+        assertEquals("Property wasn't set", optionValue, System.getProperty(IGNITE_ALLOW_START_CACHES_IN_PARALLEL));
+
+        startGrid(0).cluster().active(true);
```

Comment:
No asserts? I have the same test name in my PR https://github.com/apache/ignite/pull/5377/commits.

Reply:
+1. i.e. you can set StopNodeFailureHandler as FailureHandler in getConfiguration and check  if failureCtx is null or not.